src/components/content/heroes/aurora/aurora.tsx

```diff
@@ -112,57 +178,130 @@ export const Aurora = ({ type, className }: AuroraProps) => {
-          flareType={FlareType.LIGHT}
-          x={'20vw'}
-          y={'0vh'}
-          size={400}
-          rotation={180}
-        />
-        <Flare
-          opacity={0.5}
-          speedMultiplier={2}
-          stepSize={20}
-          flareType={FlareType.LIGHT}
-          x={'20vw'}
-          y={'90vh'}
-          size={550}
-          rotation={0}
-        />
-        <Flare
-          stepSize={20}
-          flareType={FlareType.DARK}
-          x={'50vw'}
-          y={'50vw'}
-          size={150}
-          rotation={30}
-          animationOffset={3}
-        />
-        <Flare
-          opacity={0.3}
-          stepSize={40}
-          flareType={FlareType.LIGHT}
-          x={'80vw'}
-          y={'20vw'}
-          size={250}
-          rotation={70}
-          animationOffset={7}
-        />
-        <Flare
-          stepSize={0}
-          flareType={FlareType.DARK}
-          x={'70vw'}
-          y={'300px'}
-          size={100}
-          rotation={80}
-          animationOffset={14}
-        />
+        <AuroraForeground>
+          {flaresByWeather[weather || WeatherType.NotSet]}
+          <Flare
+            key="flare-2"
+            opacity={0.5}
+            speedMultiplier={2}
+            stepSize={20}
+            flareType={FlareType.LIGHT}
+            x={'20vw'}
+            y={'90vh'}
+            size={550}
+            rotation={0}
+          />
+          ,
+          <Flare
+            key="flare-4"
+            opacity={0.3}
+            stepSize={40}
+            flareType={FlareType.LIGHT}
+            x={'80vw'}
+            y={'20vw'}
+            size={250}
+            rotation={70}
+            animationOffset={7}
+          />
+          ,
+          <Flare
+            key="flare-1"
+            opacity={0.6}
+            speedMultiplier={0.5}
+            stepSize={-80}
+            flareType={FlareType.LIGHT}
+            x={'20vw'}
+            y={'0vh'}
+            size={400}
+            rotation={180}
+          />
+          ,
+        </AuroraForeground>
       </AuroraForeground>
     </AuroraContainer>
   );
 };
+
+const flaresByWeather: { [key in WeatherType] } = {
+  [WeatherType.Sunny]: [
+    <Flare
+      key="flare-sun"
+      opacity={0.9}
+      speedMultiplier={0.1}
+      stepSize={-80}
+      flareType={FlareType.RADIAL}
+      x={'20vw'}
+      y={'40vh'}
+      size={400}
+      rotation={180}
+    />,
+    <Flare
+      key="flare-5"
+      stepSize={0}
+      flareType={FlareType.LIGHT}
+      x={'70vw'}
+      y={'300px'}
+      size={100}
+      rotation={80}
+      animationOffset={14}
+    />,
+    <Flare
+      key="flare-3"
+      stepSize={20}
+      flareType={FlareType.LIGHT}
+      x={'50vw'}
+      y={'50vw'}
+      size={150}
+      rotation={30}
+      animationOffset={3}
+    />,
+  ],
+  [WeatherType.Cloudy]: [
+    // Add your cloudy flares here
+  ],
+  [WeatherType.Rainy]: [
+    <AuroraRainyFlareBackground4 key="flare-rainy-bg-4" />,
```

Comment:
You only need reacts `key` property for elements in lists. You can savely remove them here.
I have seen them in some other occasions too, please remove them in all appearances 🧹